### PR TITLE
Db/view local sightings

### PR DESCRIPTION
**Pull request by dborg291**
**Purpose:**
When there is not a data connection for the user, their sighting info gets stored locally on the device. The user needs a way to view what sightings are local, and then be able to sync it on demand.

The of this PR was the implement the feature, and not adjust implements too much styling.

**Changes:**
1. Created `LocalSighting.jsx` and `Sighting.jsx`
2. Transitioned to using the `useAysncStorage.jsx` hook in `LocalSighting.jsx`
3. When clicking "Sync" the button is disabled for 5 seconds and shows an activity indicator. Then all the local sightings are deleted from AysncStorage. This flow is temporary until the backend is completed.

**Testing:**
Tested on both Android and iOS.

### CODE REVIEW

**Comment by ashika-m** [comment]
Could we just display all the local sightings on the home screen?

**Comment by dborg291** [comment]
> Could we just display all the local sightings on the home screen?

Good idea, I'll move it from a separate screen to on the homepage.

File: src/components/CustomDrawerContent.jsx [hunk 1]
```diff
@@ -72,6 +72,21 @@ export default function CustomDrawerContent(props) {
...
+            <Icon
+              name="archive"
+              type="material-icons"
+              color={theme.black}
+              iconStyle={styles.icon}
+            />
+            <Typography id="LOCAL_SIGHTINGS" style={styles.drawerListText} />
+          </View>
+        )}
+        onPress={() => props.navigation.navigate(screens.localSigthing)}
```
Comment (brmscheiner) [inline src/components/CustomDrawerContent.jsx:88]: typo here

File: src/screens/localSightings/LocalSightings.jsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+          setIsSyncing(false);
+        }),
+      5000
+    );
+  };
+
+  return (
+    <View style={{ flex: 1 }}>
+      <ScrollView style={{ flex: 1 }}>
+        {storedSightings != null ? (
```
Comment (brmscheiner) [inline src/screens/localSightings/LocalSightings.jsx:35]: `!==`

File: src/screens/localSightings/LocalSightings.jsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+          storedSightings.map((sighting, index) => (
+            <Sighting sighting={sighting} key={index} sightingIndex={index} />
+          ))
+        ) : (
+          <Text>
+            {' '}
+            There are currently no stighting stores locally on the device.
```
Comment (brmscheiner) [inline src/screens/localSightings/LocalSightings.jsx:42]: typo

File: src/screens/localSightings/LocalSightings.jsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  const syncSighting = () => {
+    setIsSyncing(true);
+    setTimeout(
+      () =>
+        NetInfo.fetch().then((state) => {
+          if (state.isInternetReachable) {
+            alert(
+              `Sent ${storedSightings.length} locally saved sighting(s) to the server`
+            );
+            AsyncStorage.removeItem('SightingSubmissions');
```
Comment (brmscheiner) [inline src/screens/localSightings/LocalSightings.jsx:22]: So the object gets removed from AsyncStorage, but where do they get added?

File: src/screens/home/Home.jsx [hunk 5]
```diff
@@ -43,10 +46,61 @@ const SightingCard = (props) => {
...
+  };
 
   return (
     <View style={bodyStyles.parentView}>
-      {/* TODO: Turn from ScrollView into something FlatView for performance in long term(?) */}
+      {storedSightings !== null ? (
+        <View style={offlineSightings.offlineText}>
+          <View style={{ justifyContent: 'center' }}>
+            <Text style={offlineSightings.offlineSightingsText}>
+              {' '}
```
Comment (brmscheiner) [inline src/screens/home/Home.jsx:83]: oops

File: src/screens/localSightings/LocalSightings.jsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  return (
+    <View style={{ flex: 1 }}>
+      <ScrollView style={{ flex: 1 }}>
+        {storedSightings !== null ? (
+          storedSightings.map((sighting, index) => (
+            <Sighting sighting={sighting} key={index} sightingIndex={index} />
+          ))
+        ) : (
+          <Text>
+            {' '}
```
Comment (brmscheiner) [inline src/screens/localSightings/LocalSightings.jsx:41]: oops

**Review by brmscheiner — APPROVED** [review]

**Pull request merged**